nomad/acl_endpoint.go

```diff
@@ -121,12 +121,36 @@ func (a *ACL) ListPolicies(args *structs.ACLPolicyListRequest, reply *structs.AC
 	defer metrics.MeasureSince([]string{"nomad", "acl", "list_policies"}, time.Now())
 
 	// Check management level permissions
-	if acl, err := a.srv.ResolveToken(args.AuthToken); err != nil {
+	acl, err := a.srv.ResolveToken(args.AuthToken)
+	if err != nil {
 		return err
-	} else if acl == nil || !acl.IsManagement() {
+	} else if acl == nil {
 		return structs.ErrPermissionDenied
 	}
 
+	// If it is not a management token determine the policies that may be listed
+	mgt := acl.IsManagement()
+	var policies map[string]struct{}
+	if !mgt {
+		snap, err := a.srv.fsm.State().Snapshot()
+		if err != nil {
+			return err
+		}
+
+		token, err := snap.ACLTokenBySecretID(nil, args.AuthToken)
```

Comment:
I know I left this comment somewhere last week, but I can't remember where...sorry

Is this necessary or can you reuse `acl` from above?

Reply:
Ack, found it: https://github.com/hashicorp/nomad/pull/3384#discussion_r144678912

Ignore these comments! Sorry for the noise.